Sourcery/Parsing/Composer.swift

```diff
@@ -112,8 +173,8 @@ struct Composer {
                 tupleElement.type = resolveType(typeName: tupleElement.typeName, containingType: containingType, unique: unique, modules: modules, typealiases: typealiases)
             }
         }
-        return unique[lookupName.unwrappedTypeName]
-            ?? typeFromModule(lookupName.unwrappedTypeName, modules: modules)
+
+        return unique[lookupName.unwrappedTypeName] ?? unique[lookupName.generic?.name ?? ""] ?? typeFromModule(lookupName.unwrappedTypeName, modules: modules)
```

Comment:
This still will not work if generic type comes from a different module, it should be something like this:

```swift
        let typeName = lookupName.generic?.name ?? lookupName.unwrappedTypeName
        return unique[typeName] ?? typeFromModule(typeName, modules: modules)
```